lib/lib-search/src/main/java/org/sagebionetworks/search/CloudSearchDocumentBatchBuilder.java

```diff
@@ -0,0 +1,141 @@
+package org.sagebionetworks.search;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.io.output.CountingOutputStream;
+import org.sagebionetworks.repo.model.search.Document;
+import org.sagebionetworks.util.FileProvider;
+import org.sagebionetworks.util.ValidateArgument;
+
+public class CloudSearchDocumentBatchBuilder implements Closeable {
+
+	private static final long MEBIBYTE = 1048576; // CloudSearch's Limits say MB but they really mean MiB
+	private static final long DEFAULT_MAX_SINGLE_DOCUMENT_SIZE = MEBIBYTE; //1MiB
+	private static final long DEFAULT_MAX_DOCUMENT_BATCH_SIZE = 5 * MEBIBYTE; //5MiB
+
+	static final Charset CHARSET = StandardCharsets.UTF_8;
+	static final byte[] PREFIX_BYTES = "[".getBytes(CHARSET);
+	static final byte[] SUFFIX_BYTES = "]".getBytes(CHARSET);
+	static final byte[] DELIMITER_BYTES = ",".getBytes(CHARSET);
+
+	File documentBatchFile;
+
+	CountingOutputStream countingOutputStream;
+
+	Set<String> documentIds;
+
+	//maximum bytes
+	private final long maxSingleDocumentSizeInBytes;
+	private final long maxDocumentBatchSizeInBytes;
+
+	private CloudSearchDocumentBatch builtBatch;

```

Comment:
just track the file and not the batch.  Close should delete the file if it is not null.  Build should set the file to null (in a finally) since the file has been passed to the batch and becomes the batche's responsibility.